libraries/helpers.rb

```diff
@@ -0,0 +1,189 @@
+module Php
+  module Cookbook
+    module Helpers
+      def php_conf_dir
+        if platform_family?('rhel', 'amazon')
+          '/etc'
+        else
+          "/etc/php/#{php_version}/cli"
+        end
+      end
+
+      def php_disable_mod
+        '/usr/sbin/phpdismod'
+      end
+
+      def php_enable_mod
+        '/usr/sbin/phpenmod'
+      end
+
+      def php_ext_conf_dir
+        if platform_family?('rhel', 'amazon')
+          '/etc/php.d'
+        else
+          "/etc/php/#{php_version}/mods-available"
+        end
+      end
+
+      def php_ext_dir
+        '/usr/lib64/php/modules'
+      end
+
+      def php_fpm_conf_dir
+        if platform_family?('rhel', 'amazon')
+          '/etc/php-fpm.d'
+        else
+          "/etc/php/#{php_version}/fpm"
+        end
+      end
+
+      def php_fpm_default_conf
+        if platform_family?('rhel', 'amazon')
+          '/etc/php-fpm.d/www.conf'
+        else
+          "/etc/php/#{php_version}/fpm/pool.d/www.conf"
+        end
+      end
+
+      def php_fpm_group
+        case node['platform_family']
+        when 'rhel', 'amazon'
+          'apache'
+        when 'debian'
+          'www-data'
+        end
+      end
+
+      def php_fpm_listen_group
+        case node['platform_family']
+        when 'rhel', 'amazon'
+          'apache'
+        when 'debian'
+          'www-data'
+        end
+      end
+
+      def php_fpm_listen_user
+        case node['platform_family']
+        when 'rhel', 'amazon'
+          'apache'
+        when 'debian'
+          'www-data'
+        end
+      end
+
+      def php_fpm_package
+        if platform_family?('rhel', 'amazon')
+          'php-fpm'
+        else
+          "php#{php_version}-fpm"
+        end
+      end
+
+      def php_fpm_pool_dir
+        if platform_family?('rhel', 'amazon')
+          '/etc/php-fpm.d'
+        else
+          "/etc/php/#{php_version}/fpm/pool.d"
+        end
+      end
+
+      def php_fpm_pool_cookbook
+        'php'
+      end
+
+      def php_fpm_pool_template
+        'fpm-pool.conf.erb'
+      end
+
+      def php_fpm_service
+        if platform_family?('rhel', 'amazon')
+          'php-fpm'
+        else
+          "php#{php_version}-fpm"
+        end
+      end
+
+      def php_fpm_socket
+        if platform_family?('rhel', 'amazon')
+          "/var/run/php#{php_version}-fpm.sock"
+        else
+          "/var/run/php/php#{php_version}-fpm.sock"
+        end
+      end
+
+      def php_fpm_user
+        case node['platform_family']
+        when 'rhel', 'amazon'
+          'apache'
+        when 'debian'
+          'www-data'
+        end
+      end
+
+      def php_ini_template
+        'php.ini.erb'
+      end
+
+      def php_ini_cookbook
```

Comment:
Likewise with this default value. Having it directly in the resource. Even if it's duplicated in a bunch of places. Makes it a bit simpler.